ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -218,116 +285,131 @@ export const RuntimePanel = fp.flow(withCurrentWorkspace(), withCdrVersions())((
   }
 
   return <div data-test-id='runtime-panel'>
-    <h3 style={styles.sectionHeader}>Cloud analysis environment</h3>
+    <h3 style={{...styles.baseHeader, ...styles.sectionHeader}}>Cloud analysis environment</h3>
     <div>
       Your analysis environment consists of an application and compute resources.
       Your cloud environment is unique to this workspace and not shared with other users.
     </div>
-    {/* TODO(RW-5419): Cost estimates go here. */}
-    <div style={styles.controlSection}>
-      {/* Recommended runtime: pick from default templates or change the image. */}
-      <PopupTrigger side='bottom'
-                    closeOnClick
-                    content={
-                      <React.Fragment>
-                        {
-                          fp.flow(
-                            fp.filter(['displayName', 'General Analysis']),
-                            fp.toPairs,
-                            fp.map(([i, preset]) => {
-                              return <MenuItem
-                              style={styles.presetMenuItem}
-                              key={i}
-                              onClick={() => {
-                                // renaming to avoid shadowing
-                                const {runtimeTemplate} = preset;
-                                const {presetDiskSize, presetMachineName} = fp.cond([
-                                  [() => !!runtimeTemplate.gceConfig, ({gceConfig: {bootDiskSize, machineType}}) => ({
-                                    presetDiskSize: bootDiskSize,
-                                    presetMachineName: machineType
-                                  })],
-                                  [() => !!runtimeTemplate.dataprocConfig, ({dataprocConfig: {masterDiskSize, masterMachineType}}) => ({
-                                    presetDiskSize: masterDiskSize,
-                                    presetMachineName: masterMachineType
-                                  })]
-                                ])(runtimeTemplate);
-                                const presetMachineType = fp.find(({name}) => name === presetMachineName, validLeonardoMachineTypes);
+    {switchCase(panelContent,
+      [PanelContent.Delete, () => <ConfirmDelete
+        onConfirm={() => {
+          setRuntimeStatus(RuntimeStatusRequest.Delete);
```

Comment:
Curious about your thoughts on this. Ideally, we'd trigger a loading spinner at this point, then dispatch the delete/poll for recreation. Plumbing a promise out of `setRuntimeStatus` seems problematic, and I'm not sure whether it has the right side effects.

Another idea I had was perhaps to encode the state machine in this runtimepanel render function, e.g.

```
if (deleting && runtime.status === Deleted) {
  setDeleting(false);
  setPanelContent(PanelContent.Customize);
}

return ...
  {deleting && <SpinnerOverlay />}
  ...;
```

but this also seems a bit difficult to follow.

Reply:
I also don't necessary need to wait for the runtime status to change. It may just be sufficient to wait for the `DeleteRuntime` XHR to complete. But the answers seem like they are probably the same/ similar.

Reply:
The `useRuntimeStatus` hook could return something like:
```javascript
[{runtimeStatus: runtime.status, outstandingRequest: RuntimeStatusRequest}, setRuntimeStatus]
``` 
The hook would then update and trigger a render when a new state is requested, or when the store gets an update from the state. Do you think this would be too complex, or too much behavior for the hook to handle?

Reply:
Hmm interesting. I think this could work - I don't think the complexity of the hook itself would be too much here. My main concern is that handling of the post-delete transition seems a little sketchy in any scenario where I can't write a straight-line: "await x; then do y". For example, adapting your suggestion I think I'd need something like this:


```
// Assume that there was previously an outstanding request: since we're now deleting, we assume the pending deletion
// request has completed and we initiate navigation away. 
if (!outstandingRequest && runtime.status === Deleting) {
  setPanelContent(PanelContent.Customize);
}

return ...
  {oustandingRequest && <SpinnerOverlay />}
  ...;

```

It's not awful, but I am a bit worried about the substantial decoupling the navigation from the original click.

Reply:
I'll reply to your comment there, but lets completely separate that from this discussion. Here I'm trying to do something very simple: show a "loading" spinner while the DeleteRuntime `fetch()` is still pending, then change the panel after that completes.